nifi-nar-bundles/nifi-standard-services/nifi-distributed-cache-services-bundle/nifi-distributed-cache-server/src/main/java/org/apache/nifi/distributed/cache/server/SetCacheServer.java

```diff
@@ -102,4 +100,18 @@ protected void finalize() throws Throwable {
         }
     }
 
+    private byte[] readValue(final DataInputStream dis) throws IOException {
+        final int numBytes = validateInt(dis.readInt(), getMaxReadSize(), "readValue():");
+        final byte[] buffer = new byte[numBytes];
+        dis.readFully(buffer);
+        return buffer;
+    }
+
+    private int validateInt(final int value, final int max, final String identifier) throws IOException {
+        if (value <= max) {
+            return value;
+        } else {
+            throw new IOException(new IllegalArgumentException(identifier + value));
+        }
+    }
```

Comment:
As mentioned above, recommend removing the `identifier` parameter and adjusting the exception message.  The wrapping `IOException` also seems unnecessary.
```suggestion
    private int validateSize(final int size) {
        if (size <= getMaxReadSize()) {
            return size;
        } else {
            throw new IllegalStateException(String.format("Size [%d] exceeds maximum configured read [%d]", size, getMaxReadSize()));
        }
    }
```